### PR TITLE
count no. of bits in a number added

### PR DESCRIPTION
**Pull request by shivammahajancse**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(#255 )

### CODE REVIEW

**Comment by shivammahajancse** [comment]
Please review sir @HarshCasper @iamrajiv @ankitaggarwal23 @atarax665 @gat786

File: docs/mathematics/count-bits.mdx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+     if(a[i]==x)
+     {
+       count++;
+     }
+   }
+   return count;
+}
+
+//main function
+int main()
```
Comment (vybhav72954) [inline docs/mathematics/count-bits.mdx:66]: The main function shall only have the driver code

File: docs/mathematics/count-bits.mdx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+4
+```
+
+## Complexity Analysis
+ 
+- **Time Complexity:** `O(logn)`
+
+- **Space Complexity:** `O(1)`
+
+where `n` is the number of elements.
```
Comment (vybhav72954) [inline docs/mathematics/count-bits.mdx:136]: Credits?

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

File: docs/mathematics/count-bits.mdx (hunk 2 above)
Comment (shivammahajancse) [reply docs/mathematics/count-bits.mdx:136]: Sir code is written by me .

**Comment by shivammahajancse** [comment]
All requested changes done sir @vybhav72954

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@shivammahajancse Resolve the merge conflict.

**Review by iamrajiv — APPROVED** [review]

File: docs/mathematics/count-bits.mdx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+Binary representation of given number(6)=110
+So number of bits in 6 is 3
+Print 3
+```
+## Algorithm
+
+For count number of bits in a number, we follow the following algorithm:
+
+* First, we convert the number into its binary representation.
+
```
[the remark this answers is not in the document]
Comment (atarax665) [reply docs/mathematics/count-bits.mdx:27]: @shivammahajancse You haven't resolved this yet.

[the remark this answers is not in the document]
Comment (shivammahajancse) [reply docs/mathematics/count-bits.mdx]: @atarax665 Sir done

**Review by tk315 — APPROVED** [review]

**Review by siddharth25pandey — APPROVED** [review]
@shivammahajancse LGTM 💯 Able to merge

**Pull request merged**